### DISCUSSION TITLE
v2.4.0

### DISCUSSION THREAD

**Discussion by mikecao** [umami-software/umami#2158] [General]
## Features
- The event data screen now shows the related event
 
![image](https://github.com/umami-software/umami/assets/477975/0f7808b7-a932-4c3a-ba20-f67280fa513e)

You can click on the event to see the saved values

![image](https://github.com/umami-software/umami/assets/477975/0f6a481b-2288-49ad-acfe-069383808e9b)

- Websites from teams now appear on the dashboard
- Added ability to delete reports
- Added new environment variable `DEFAULT_LOCALE` which will set the default locale of the application other than `en-US`

## Fixes
- Fixed new release message throwing an error
- Fixed funnel report for MySQL
- Fixed check for updates process
- Fixed `BASE_PATH` not working for share URLs
- Fixed tracker sending wrong event data payload
- Fixed toggle charts not working on dashboard
- Fixed missing `(None)` referrers
- Fixed website nav menu for mobile
- Fixed realtime log not showing pageviews

## Changes
- Upgraded to Prisma v5.0.0
- Updated language translations

A huge thanks to all the contributors on this release! @AkashRajpurohit @mikusaa @RikoDEV @yumusb @Maxime-J @franciscao633 @tairosonloa @wangel13 @h0ek @nailuoGG @briancao @0xflotus @humbertleonardo 

<hr /><em>This discussion was created from the release <a href='https://github.com/umami-software/umami/releases/tag/v2.4.0'>v2.4.0</a>.</em>

**Comment by 4ortytwo** [comment]
After this update, my self-hosted Postgresql Umami on DO won't display any event data.

Event Data on the main website view is visible, though. 

<img width="1264" alt="image" src="https://github.com/umami-software/umami/assets/44406493/337943fc-458d-4329-bc1b-7896dd6e42a4">

**Reply by uncenter** [reply]
Same for me.

**Reply by humbertleonardo** [reply]
Have you tried running "yarn build" and "pm2 restart 0 --update-env"?

**Reply by franciscao633** [reply]
This is fixed in dev. We will do a hot fix tonight/tomorrow

**Comment by skipperbent** [comment]
Event data page doesn't work for non-admins: fails to retrieve data (shows fields and data total records fine) but no data available.

`/api/event-data/events?websiteId=382d2dcd-1358-46b0-a6cf` -> 401 Unauthorized

What's the purpose of events if there is no way of filtering them? I mean i can see the same stats elsewhere by day, month, year so why make an events page that shows the exact same information? I fail to see the purpose of the event page and events in general in Umami given the data can't be retrieved in any way.

**Reply by uncenter** [reply]
The data can be retrieved, you just need to authorize with an access token. 

> First you need to get a token in order to make API requests.

(https://umami.is/docs/authentication)

**Reply by skipperbent** [reply]
I'm not using the API manually I get the console error whenever I access the event page from the administration. Tried logging in and out again didn't solve it so I guess the view or non admin user might not have access to event data for some reason? Event data shows up fine on other pages like live view etc. Not sure how to debug it further

**Reply by uncenter** [reply]
Gotcha I was just confused why you were showing a /api URL. Both of the events pages start with `https://<domain>/websites/<id>/` if you are using the dashboard.

**Reply by skipperbent** [reply]
Sorry for not being clear enough and for the confusion 🙂
